.expeditor/scripts/finish_release/bump_rustfmt.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This script attempts to find the latest installable nightly rustfmt.
+# When found, it bumps the repo rustfmt version and submits a new PR
+# with the new version pin and any code base formatting changes from the
+# new version.
+
+# Assumptions:
+# 1. This script runs on a x86_64-unknown-linux-gnu target
+# 2. If rustfmt is installable on x86_64-unknown-linux-gnu, the other Tier1
+#    platforms are assumed to also have a working version.
+
+set -Eeuo pipefail
+
+# shellcheck source=.expeditor/scripts/verify/shared.sh
+source .expeditor/scripts/verify/shared.sh
+
+TODAY=$(date '+%Y-%m-%d')
+# pick out the date from a string like: nightly-2019-12-04
+CURRENT_RUSTFMT_DATE=$(awk -F- '{print $2"-"$3"-"$4}' RUSTFMT_VERSION)
+# convert to seconds for easier date comparison later
+CURRENT_RUSTFMT_DATE_S=$(date '+%s' --date "${CURRENT_RUSTFMT_DATE}")
+CHANNEL=nightly
+# a maximum iterator value for searching elligible nightlies
+MAX_ATTEMPTS=90
+
+# if set, this will be the next version
+export FOUND_VERSION
+
+install_rustup
+
+# set minimal profile so only core items + rustfmt are installed
+echo "--- :habicat: Setting minimal profile"
+rustup set profile minimal
+
+for ((i=0; i<MAX_ATTEMPTS; i++)); do
+  # begin searching nightly date versions using current date
+  # as a starting point
+  attempt_date=$(date '+%Y-%m-%d' --date "${TODAY} -${i} day")
+  # use the conversion to seconds for easy date comparison
+  attempt_date_s=$(date '+%s' --date "${attempt_date}")
+  attempt_version="${CHANNEL}-${attempt_date}"
+  echo "--- Trying Nightly Date Version: ${attempt_version}"
+  if rustup toolchain install "${attempt_version}" --component rustfmt; then
+    FOUND_VERSION=${attempt_version}
+    break
+  fi
+  if [ "${attempt_date_s}" -le "${CURRENT_RUSTFMT_DATE_S}" ]; then
+    # Exit early since there is no newer version available than what is
+    # currently pinned in RUSTFMT_VERSION file.
+    echo "--- No newer ${CHANNEL} versions found. Nothing to do."
+    exit 0
+  fi
+done
+
+if [ -z "${FOUND_VERSION}" ]; then
+  # Something is likely wrong if we did not find an installable version.
+  echo "--- :thumbsdown: No candidate versions found within ${MAX_ATTEMPTS}!"
+  exit 1
```

Comment:
I think we probably don't want to fail the entire post-release pipeline if this fails. Making it a soft fail would be OK; notifying in Slack could be another thing to add, but might be overkill.

Reply:
Good callout. I've addressed in https://github.com/habitat-sh/habitat/pull/7258/commits/08d43ea91b07ee61f280206185d8e01c2d2af06e